socs/agents/lakeshore240/agent.py

```diff
@@ -1,82 +1,139 @@
 import argparse
 import os
+import queue
 import time
+import traceback
 import warnings
-from typing import Optional
+from dataclasses import dataclass
+from typing import Any, Dict, Generator, Optional, Tuple
 
-import txaio
+import txaio  # type: ignore
+from ocs import ocs_agent, site_config
+from ocs.ocs_twisted import Pacemaker
 
 from socs.Lakeshore.Lakeshore240 import Module
 
-on_rtd = os.environ.get('READTHEDOCS') == 'True'
-if not on_rtd:
-    from ocs import ocs_agent, site_config
-    from ocs.ocs_twisted import TimeoutLock
+txaio.use_twisted()
 
 
-class LS240_Agent:
+on_rtd = os.environ.get("READTHEDOCS") == "True"
+if not on_rtd:
+    log = txaio.make_logger()  # pylint: disable=E1101
 
-    def __init__(self, agent, port="/dev/ttyUSB0", f_sample=2.5):
 
-        self.agent: ocs_agent.OCSAgent = agent
-        self.log = agent.log
-        self.lock = TimeoutLock()
+ActionReturnType = Optional[Dict[str, Any]]
+OcsOpReturnType = Tuple[bool, str]
+OcsInlineCallbackReturnType = Generator[Any, Any, OcsOpReturnType]
 
-        self.port = port
-        self.module: Optional[Module] = None
 
-        self.f_sample = f_sample
+class Actions:
+    "Namespace to hold action classes for the Lakeshore240 agent."
 
-        self.initialized = False
-        self.take_data = False
+    class BaseAction:
+        "Base class for all actions."
 
-        # Registers Temperature and Voltage feeds
-        agg_params = {
-            'frame_length': 60,
-        }
-        self.agent.register_feed('temperatures',
-                                 record=True,
-                                 agg_params=agg_params,
-                                 buffer_time=1)
+        def __post_init__(self) -> None:
+            self.processed: bool = False
+            self.success: bool = False
+            self.traceback: Optional[str] = None
+            self.result: Any = None
```

Comment:
If this structure ends up sticking around these should be private, else agent authors need to be sure not to collide any agent arguments with these (or any future) attributes within the `BaseAction` -- I imagine once this is more solidified `BaseAction` and similar machinery would move to ocs-core.